prov/shm/src/smr_ep.c

```diff
@@ -1329,6 +1329,13 @@ static int smr_ep_ctrl(struct fid *fid, int command, void *arg)
 			}
 		}
 
+		/* TODO: When we add support ipc support to other HMEM providers,
+		 * include them in this check */
+		if (rocr_is_ipc_enabled())
```

Comment:
I'd still prefer a generic check here.  We're overriding smr_progress_ipc_list(), which is a generic function not specific to rocr.  I'd rather see something like:

```
if (ofi_hmem_any_ipc_enabled())
    blah blah
```
And have that call make the specific checks that are needed.  And the abstracted call we should use is ofi_hmem_is_ipc_enabled(), not a direct call to rocr.
```
bool ofi_hmem_any_ipc_enabled(void)
{
    int iface;

    for (iface = 0; iface < ARRAY_SIZE(hmem_ops); iface++)
        if (ofi_hmem_is_ipc_enabled(iface))
            return true;
    return false;
}
```
and then adding support for other async calls doesn't require any changes through this path.